README.rst

```diff
@@ -26,3 +26,371 @@ Run `tox` to run the whole test suite:
+   # shell interaction instance
+   shell = ShellInteraction(ctrl)
+   shell.start_term()               # start a serial terminal
+   print(shell.cmd("help"))         # print the command result
+   shell.stop_term()                # close the terminal
+
+or using the already provided ``Help`` ``ShellInteraction``:
+
+.. code:: python
+
+   class Help(ShellInteraction):
+       """Help ShellInteraction"""
+       @ShellInteraction.check_term
+       def help(self, timeout=-1, async_=False):
+           """Sends the reboot command via the terminal"""
+           return self.cmd("help", timeout, async_)
+
+.. code:: python
+
+   from riotctrl.ctrl import RIOTCtrl
+   from riotctrl_shell.sys import Help
+
+   env = {'BOARD': 'native'}
+   # if not running from the application directory the a path must be provided
+   ctrl = RIOTCtrl(env=env, application_directory='.')
+   # flash the application
+   ctrl.flash()                     # alias for ctrl.make_run(['flash'])
+   # shell interaction instance, Help uses the @ShellInteraction.check_term
+   # decorator, it will start the terminal if its not yet running, and close
+   # it after the command ends
+   shell = Help(ctrl)              #
+   print(shell.help())             # print the command result
+
+Writing SAUL ShellInteraction
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+``examples/saul`` has a pretty simple shell:
+
+::
+
+   > saul help
+   saul help
+   usage: saul read|write
+   > saul read
+   saul read
+   usage: saul read <device id>|all
+   > saul write
+   saul write
+   usage: saul write <device id> <value 0> [<value 1> [<value 2]]
+   > saul
+   saul
+   ID  Class       Name
+   #0  SENSE_XX    x1
+   #1  SENSE_XX    x2
+
+Basically four commands a shell interaction for listing devices can be
+written as:
+
+.. code:: python
+
+   from riotctrl.shell import ShellInteraction
+
+
+   class SaulShell(ShellInteraction):
+       @ShellInteraction.check_term
+       def saul_read(self, timeout=-1, async_=False):
+           return self.cmd("saul", timeout, async_)
+
+Since all commands will share the same start: ``saul``, we can write
+base same all on a basic ``saul`` cmd with extra args:
+
+.. code:: python
+
+   from riotctrl.shell import ShellInteraction
+
+
+   class SaulShell(ShellInteraction):
+       @ShellInteraction.check_term
+       def saul_cmd(self, args=None, timeout=-1, async_=False):
+           cmd = "saul"
+           if args is not None:
+               cmd += " {args}".format(args=" ".join(str(a) for a in args))
+           return self.cmd(cmd, timeout=timeout, async_=False)
+
+       def saul_read(self, dev_id="all", timeout=-1, async_=False):
+           return self.saul_cmd(args=("read", f"{dev_id}",), timeout=timeout, async_=async_)
+
+       def saul_help(self, timeout=-1, async_=False):
+           return self.saul_cmd(args=("help",), timeout=timeout, async_=async_)
+
+Extending with ``saul_write`` command is left as an exercise.
```

Comment:
I've heard people hate this if sentences like these appear in _documentation_ ;-).

Reply:
ok, removed